src/hivpy/hiv_status.py

```diff
@@ -250,6 +267,172 @@ def stp_HIV_transmission(self, person):
+                                                      use_size=True,
+                                                      sub_pop=people_with_nonmonogamous_ltp)
+        population.set_present_variable(col.LTP_STATUS, partner_infected, people_with_nonmonogamous_ltp)
+
+        # Monogamous Partner Case: subjects infect partners
+        def calculate_infected_ltp_monogamous(sex, age_group, vl_group, sti, size):
+            risk_to_ltp = rng.normal(
+                self.transmission_rate_means[vl_group],
+                self.transmission_rate_sigmas[vl_group],
+                size=size
+            )
+            if (sex == SexType.Male):
+                if (age_group == 0):
+                    risk_to_ltp *= self.young_women_transmission_factor
+                else:
+                    risk_to_ltp *= self.women_transmission_factor
+                if sti:
+                    risk_to_ltp *= self.sti_transmission_factor
+
+            return rng.uniform(0, 1, size=size) < risk_to_ltp
+
+        people_with_monogamous_ltp_and_hiv = population.get_sub_pop([(col.LONG_TERM_PARTNER, op.eq, True),
+                                                                    (col.LTP_MONOGAMOUS, op.eq, True),
+                                                                    (col.HIV_STATUS, op.eq, True)])
+        partner_infected = population.transform_group([col.SEX, col.AGE_GROUP, col.VIRAL_LOAD_GROUP, col.STI],
+                                                      calculate_infected_ltp_monogamous,
+                                                      use_size=True,
+                                                      sub_pop=people_with_monogamous_ltp_and_hiv)
+        population.set_present_variable(col.LTP_STATUS, partner_infected, people_with_monogamous_ltp_and_hiv)
+
+    def prob_of_infection_from_infected_ltp(self, population: Population):
+        """
+        Sets the probability of infecttion (RISK_LTP)for the population set
+          for which infection occurs from an infected long term partner.
+        """
+        def calculate_risk_of_infection(viral_suppression, ltp_status, size):
+            subgroup = population.get_sub_pop([(col.VIRAL_SUPPRESSION, op.eq, viral_suppression),
+                                               (col.LTP_STATUS, op.eq, ltp_status)])
+            if viral_suppression:
+                risk = rng.normal(self.tr_rate_undetectable_vl, 0.000025, size)
+                vlg = 0
+            else:
+                if ltp_status:
+                    risk = rng.normal((0.05*self.transmission_factor), 0.0125, size)
+                    vlg = 3
+                else:
+                    risk = rng.normal(self.tr_rate_primary, 0.075, size)
+                    vlg = 5
+                    population.set_present_variable(col.LTP_IN_PRIMARY, True, subgroup)
+
+            population.set_present_variable(col.RISK_LTP, risk, subgroup)
```

Comment:
We can't use `RISK_LTP` because this column is used for something else in sexual behaviour 
(RISK_LTP is the change in risk behaviour for a person with a long term partner)

Reply:
modified to `RIK_LTP_INFECTED`